src/Servers/Kestrel/Core/src/Internal/AddressBinder.cs

```diff
@@ -141,6 +141,28 @@ internal static ListenOptions ParseAddress(string address, out bool https)
         }
 
         return options;
+
+        bool IsLocalhost(string host, out string? prefix)
```

Comment:
`parsedAddress.Host` and the `host` are the same, are they just used in an interchangeable manner, right? One is captured by local scope, it seems `host` parameter would be a better option to use then. If that is the case, also in line 156 `!string.Equals` could become a length check something `host.Length > 10` ?